### ISSUE TITLE
OpenSSL BytesToKey() for embedded system

### ISSUE THREAD

**Issue by jason-smith-01** [openssl/openssl#21402]
In evp.h, `BytesToKey()` generates the key from the passhphrase, IV and the MD5 digest in order to decrypt information used by the public/private keypair generation using AES-256-CBC. I am trying to isolate all the EVP_* functions into a standalone implementation for an embedded system. Does anyone have the actual function or what exactly it does?

KEY DERIVATION ALGORITHM The key and IV is derived by concatenating D_1, D_2, etc until enough data is available for the key and IV. D_i is defined as:

`D_i = HASH^count(D_(i-1) || data || salt)`

**Comment by jason-smith-01** [comment]
1st issue I solved as that was user-issue. In the implementation, I mis-calculated the key size being 128 bytes from 1024-bit encryption. When I tried using 2048-bit encryption, I forgot to add an additional 128 byte to the key (using key and iv for encryption/decryption)

2nd issue: sort of solved.
I've discovered that I can create 2048-bit keys with the traditional method and asn1parse actually works with these keys but not 1024-bit keys. so maybe that is an issue?
